File: src/Unit-1/DoThis/Messages.cs
namespace WinTail
{
    public class Messages
    {
        #region Neutral / System messages

        public class ContinueProcessing
        {

        }

        #endregion

        #region Success messages

        public class InputSuccess
        {
            public string Reason { get; private set; }

            public InputSuccess(string reason)
            {
                this.Reason = reason;
            }
        }

        #endregion

        #region Failure messages

        public class InputError
        {
            public string Reason { get; private set; }

            public InputError(string reason)
            {
                this.Reason = reason;
            }
        }

        public class NullInputError : InputError
        {
            public NullInputError(string reason)
                : base(reason)
            {
            }
        }

        public class ValidationInputError : InputError
        {
            public ValidationInputError(string reason)
                : base(reason)
            {
            }
        }


        #endregion

    }
}

File: src/Unit-1/FSharp/DoThis/Messages.fs
module WinTail.Messages

open Akka.Actor

type ErrorType =
| Null
| Validation

type InputResult =
| InputSuccess of string
| InputError of reason:string * errorType:ErrorType

type TailCommand =
| StartTail of filePath:string * reporterActor:IActorRef
| StopTail of filePath:string

type FileCommand =
| FileWrite of fileName:string
| FileError of fileName:string * reason:string
| InitialRead of fileName:string * text:string